rosbag2_tests/test/rosbag2_tests/record_fixture.hpp

```diff
@@ -45,18 +45,22 @@ class RecordFixture : public TemporaryDirectoryFixture
 public:
   RecordFixture()
   {
-    root_bag_path_ = (rcpputils::fs::path(temporary_dir_path_) / "bag").string();
-    storage_path_ = (rcpputils::fs::path(root_bag_path_) / "bag_0").string();
-    database_path_ = storage_path_ + ".db3";
-    std::cout << "Database " << database_path_ << " in " << temporary_dir_path_ << std::endl;
+    set_path("bag");
   }
 
-  void SetUp() override
+  void set_path(const std::string & bagname)
   {
-    const auto path = rcpputils::fs::path{root_bag_path_};
-    if (rcpputils::fs::exists(path)) {
-      remove_directory_recursively(path.string());
+    auto root_path = rcpputils::fs::path{temporary_dir_path_} / bagname;
+
+    if (root_path.exists()) {
+      remove_directory_recursively(root_path.string());
```

Comment:
Shouldn't the removal happen in TearDown or at the end of each test that called set_path? Seems strange to check if resources exist that weren't created by a brand new test (RAII).

Reply:
I'm addressing this in the source as well.

The test fixture already does define an RAII cleanup on TestFixture dtor. This is ensuring that the path specified is empty.

Reply:
+1 to Devin's comment, can't we do the clean up in `SetUp` ?

Reply:
I'm switching to `ASSERT_FALSE` on exist since cleanup should already be handled.